packages/cinterop/src/commonMain/kotlin/io/realm/interop/RealmEnums.kt

```diff
@@ -65,3 +103,92 @@ expect enum class SchemaValidationMode {
     RLM_SCHEMA_VALIDATION_SYNC,
     RLM_SCHEMA_VALIDATION_REJECT_EMBEDDED_ORPHANS,
 }
+
+expect enum class ErrorType {
+    RLM_ERR_NONE,
+    RLM_ERR_UNKNOWN,
+    RLM_ERR_OTHER_EXCEPTION,
+    RLM_ERR_OUT_OF_MEMORY,
+    RLM_ERR_NOT_CLONABLE,
+    RLM_ERR_NOT_IN_A_TRANSACTION,
+    RLM_ERR_WRONG_THREAD,
+    RLM_ERR_INVALIDATED_OBJECT,
+    RLM_ERR_INVALID_PROPERTY,
+    RLM_ERR_MISSING_PROPERTY_VALUE,
+    RLM_ERR_PROPERTY_TYPE_MISMATCH,
+    RLM_ERR_MISSING_PRIMARY_KEY,
+    RLM_ERR_UNEXPECTED_PRIMARY_KEY,
+    RLM_ERR_WRONG_PRIMARY_KEY_TYPE,
+    RLM_ERR_MODIFY_PRIMARY_KEY,
+    RLM_ERR_READ_ONLY_PROPERTY,
+    RLM_ERR_PROPERTY_NOT_NULLABLE,
+    RLM_ERR_INVALID_ARGUMENT,
+    RLM_ERR_LOGIC,
+    RLM_ERR_NO_SUCH_TABLE,
+    RLM_ERR_NO_SUCH_OBJECT,
+    RLM_ERR_CROSS_TABLE_LINK_TARGET,
+    RLM_ERR_UNSUPPORTED_FILE_FORMAT_VERSION,
+    RLM_ERR_MULTIPLE_SYNC_AGENTS,
+    RLM_ERR_ADDRESS_SPACE_EXHAUSTED,
+    RLM_ERR_MAXIMUM_FILE_SIZE_EXCEEDED,
+    RLM_ERR_OUT_OF_DISK_SPACE,
+    RLM_ERR_KEY_NOT_FOUND,
+    RLM_ERR_COLUMN_NOT_FOUND,
+    RLM_ERR_COLUMN_ALREADY_EXISTS,
+    RLM_ERR_KEY_ALREADY_USED,
+    RLM_ERR_SERIALIZATION_ERROR,
+    RLM_ERR_INVALID_PATH_ERROR,
+    RLM_ERR_DUPLICATE_PRIMARY_KEY_VALUE,
+    RLM_ERR_INDEX_OUT_OF_BOUNDS,
+    RLM_ERR_INVALID_QUERY_STRING,
+    RLM_ERR_INVALID_QUERY,
+    RLM_ERR_CALLBACK;
+}
+
+/**
+ * Maps core error types to their Kotlin exception representations.
+ */
+@Suppress("ComplexMethod")
+fun errorTypeToThrowable(errorType: ErrorType, message: String?): Throwable {
+    return when (errorType) {
```

Comment:
Still trying to think if doing this mapping is a good idea or if it just introduces an extra layer for no particular reason 🤔 .

I can see specific exceptions being easier to catch in the library layer though, and any unhandled case will just bubble up. While if we just have e.g `RealmCoreException(ErrorType)` then all call sites will need to do a switch for the type they want to handle.

So I'm probably leaning towards this being fine, but would like others opinion on this as well.